src/Microsoft.DotNet.Interactive/Parsing/SubmissionParser.cs

```diff
@@ -476,6 +511,17 @@ static bool ContainsInvalidCharactersForValueReference(ReadOnlySpan<char> tokenT
             }
         }
 
+        private static string GetTypeHint(Symbol symbol) =>
+            symbol switch
+            {
+                IValueDescriptor<DateTime> => "datetime-local",
+                IValueDescriptor<int> => "number",
```

Comment:
Curious, how did we populate this list and how do we keep it updated with new types? I'm sure it's not the end of the world if this doesn't cover everything as well.

Reply:
#2245 makes this mechanism extensible.